### PR TITLE
Flux strategy review

### PR DESCRIPTION
**Pull request by naddison36**
* Fixed deploy script to use Timelock contract as governor
* Gas optimizations
  * 16.6k saving on withdrawAll
* Natspec updates
* Added more fork tests

Gas comparisons were done by running

```
FORK_BLOCK_NUMBER=17873427 yarn test:fork ./test/strategies/flux.fork-test.js
```

Gas usage before changes
```
|  Contract           ·  Method                     ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
······················|·····························|·············|·············|·············|···············|··············
|  AaveStrategy       ·  withdraw                   ·     193684  ·     213674  ·     205772  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  ERC20              ·  approve                    ·      26174  ·      60107  ·      40648  ·           72  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  ERC20              ·  transfer                   ·      41333  ·      88211  ·      60651  ·           92  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  Flipper            ·  withdrawAll                ·          -  ·          -  ·     521110  ·            1  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  FluxStrategyProxy  ·  initialize                 ·          -  ·          -  ·     369124  ·            2  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  Governor           ·  execute                    ·     167465  ·     518536  ·     323534  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  Governor           ·  queue                      ·     173380  ·     422199  ·     265477  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  MockVault          ·  allocate                   ·      66795  ·    1463853  ·     726385  ·           16  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  MockVault          ·  mint                       ·    1142575  ·    1707704  ·    1607171  ·            8  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  MockVault          ·  rebase                     ·          -  ·          -  ·    1061687  ·            1  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  OETHVault          ·  setAssetDefaultStrategy    ·          -  ·          -  ·      58439  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  OETHVault          ·  withdrawAllFromStrategies  ·          -  ·          -  ·    4792818  ·            1  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  OracleRouter       ·  cacheDecimals              ·      52836  ·      53123  ·      52989  ·            7  ·          -  │
```

Gas usages after the changes
```
|  Contract           ·  Method                     ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
······················|·····························|·············|·············|·············|···············|··············
|  AaveStrategy       ·  withdraw                   ·     193684  ·     213674  ·     205772  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  ERC20              ·  approve                    ·      26174  ·      60107  ·      40648  ·           72  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  ERC20              ·  transfer                   ·      41333  ·      88211  ·      60651  ·           92  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  Flipper            ·  withdrawAll                ·          -  ·          -  ·     504413  ·            1  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  FluxStrategyProxy  ·  initialize                 ·          -  ·          -  ·     369124  ·            2  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  Governor           ·  execute                    ·     167465  ·     518536  ·     323534  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  Governor           ·  queue                      ·     173380  ·     422199  ·     265477  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  MockVault          ·  allocate                   ·      66795  ·    1463853  ·     726385  ·           16  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  MockVault          ·  mint                       ·    1142575  ·    1707704  ·    1607171  ·            8  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  MockVault          ·  rebase                     ·          -  ·          -  ·    1061687  ·            1  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  OETHVault          ·  setAssetDefaultStrategy    ·          -  ·          -  ·      58439  ·            3  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  OETHVault          ·  withdrawAllFromStrategies  ·          -  ·          -  ·    4792561  ·            1  ·          -  │
······················|·····························|·············|·············|·············|···············|··············
|  OracleRouter       ·  cacheDecimals              ·      52836  ·      53123  ·      52989  ·            7  ·          -  │
```

### CODE REVIEW

File: contracts/test/strategies/flux.fork-test.js [hunk 1]
```diff
@@ -94,6 +127,39 @@ forkOnlyDescribe("Flux strategy", function () {
       expect(vaultUsdtDiff).to.approxEqualTolerance(usdtUnits, 1);
     });
   });
+
+  describe("Administration", function () {
```
Comment (shahthepro) [inline contracts/test/strategies/flux.fork-test.js:131]: Thanks for adding these but I believe these are better suited in unit tests

**Review by shahthepro — APPROVED** [review]
Awesome, LGTM :)

File: contracts/contracts/strategies/CompoundStrategy.sol [hunk 2]
```diff
@@ -194,17 +195,17 @@ contract CompoundStrategy is BaseCompoundStrategy {
...
+     * @notice Approve the spending of all assets by their corresponding cToken,
      *      if for some reason is it necessary.
      */
     function safeApproveAllTokens() external override {
         uint256 assetCount = assetsMapped.length;
-        for (uint256 i = 0; i < assetCount; i++) {
-            address asset = assetsMapped[i];
-            address cToken = assetToPToken[asset];
+        for (uint256 i = 0; i < assetCount; ++i) {
+            IERC20 asset = IERC20(assetsMapped[i]);
```
Comment (sparrowDom) [inline contracts/contracts/strategies/CompoundStrategy.sol:204]: Just curious, this sort of cast isn't a gas saving probably just nicer to read?

Comment (naddison36) [reply contracts/contracts/strategies/CompoundStrategy.sol:204]: casting doesn't cost any gas. It was more consistent with other parts of the code.

File: contracts/test/strategies/flux.fork-test.js (hunk 1 above)
Comment (naddison36) [reply contracts/test/strategies/flux.fork-test.js:131]: The problem is the unit tests use mock tokens so it's nice to test reapprovals can be done on a fork.
It also tests the governor access has been wired up correctly.

**Review by sparrowDom — APPROVED** [review]
LGTM

**Pull request merged**